apps/advisor/pkg/app/app.go

```diff
@@ -23,6 +55,21 @@ func New(cfg app.Config) (app.App, error) {
 		ManagedKinds: []simple.AppManagedKind{
 			{
 				Kind: advisorv0alpha1.CheckKind(),
+				Validator: &simple.Validator{
+					ValidateFunc: func(ctx context.Context, req *app.AdmissionRequest) error {
+						_, err := getCheck(req.Object, checkMap)
+						return err
+					},
+				},
+				Watcher: &simple.Watcher{
+					AddFunc: func(ctx context.Context, obj resource.Object) error {
+						check, err := getCheck(obj, checkMap)
+						if err != nil {
+							return err
+						}
+						return processCheck(ctx, client, obj, check)
```

Comment:
Just want to confirm a few (long term) things:
- Do we plan to cache the result of a check instead of always evaluating? 
- Other than the public API, will there be some service that tries to evaluate checks regularly (so the user doesn't have to always manually do it)?